docs/docs/security.md

```diff
@@ -1,167 +1,162 @@
+there is no direct access to the OS. Enclaves can communicate with the OS only by exchanging messages through the
+host code.
 
-But an enclave is just an ordinary piece of code running in an ordinary program. That means it cannot directly access
-hardware. Like any other program it has to ask the kernel to handle requests on its behalf, because the kernel is the
-only program running in the CPU's supervisor mode where hardware can be controlled ("ring 0" on Intel architecture chips).
-In fact, inside an enclave there is no direct access to the operating system at all. The only thing you can do is
-exchange messages with the host code. The host may choose to relay your requests through the kernel to the hardware
-honestly, or any of those components (host software, kernel, the hardware itself) may have been modified maliciously.
-The enclave protections are a feature of the *CPU*, not the entire computer, and the CPU maintains state only until
-it is reset or the power is lost.
+If any of the components (host software, kernel, the hardware itself) are malicious, there is a security
+issue when enclaves access hardware through exchanging messages with the host. Remember that enclave protections are a
+feature of the *CPU*, not the entire computer.
 
-You can think of this another way. If the enclave stores some data to disk, nothing stops the owner of the computer
+You can think of this another way. If the enclave stores data to disk, nothing stops the owner of the computer from
 stopping the enclave host process and then editing the files on disk.
 
-Because enclaves can generate encryption keys private to themselves, encryption and authentication can be used to stop
-the host editing the data. Data encrypted in this way is called **sealed data**. Sealed data can be re-requested from
-the operating system and decrypted inside the enclave.
+You can encrypt and authenticate data using the private keys generated by enclaves to stop the host from editing it.
+Data encrypted in this way is called **sealed data**. Sealed data can be re-requested from the OS and
+decrypted inside the enclave. Conclave handles the sealing process for you.
+
+However, there's one class of attack encryption cannot stop. This attack is when the host gives you back older data than
+was requested. The owner of the computer controls the system clock, which makes it unreliable. Additionally,
+the owner can kill and restart the host process whenever they like. This means that a system owner can tamper with 
+the enclave's sense of time and order of events. By snapshotting the stored (sealed, encrypted) data an enclave has 
+provided after each network message from a client is delivered, the enclave can be "rewound" to any
+point. Then, stored messages from the clients can be replayed back to the enclave in different orders, or with some
+messages dropped.
 
+Such attacks are called memento/rewind/replay attacks. You need to consider this type of attack while designing your
```

Comment:
Update this to mention the existing feature we have to mitigate this attack and link to relevant section in persistence.md.

Reply:
Updated.